src/main/java/net/sf/jabref/exporter/SaveDatabaseAction.java

```diff
@@ -358,25 +358,23 @@ public void saveAs() throws Throwable {
             }
         }
 
-        if (f != null) {
-            File oldFile = panel.getBibDatabaseContext().getDatabaseFile();
-            panel.getBibDatabaseContext().setDatabaseFile(f);
-            Globals.prefs.put(JabRefPreferences.WORKING_DIRECTORY, f.getParent());
-            runCommand();
-            // If the operation failed, revert the file field and return:
-            if (!success) {
-                panel.getBibDatabaseContext().setDatabaseFile(oldFile);
-                return;
-            }
-            // Register so we get notifications about outside changes to the file.
-            try {
-                panel.setFileMonitorHandle(Globals.fileUpdateMonitor.addUpdateListener(panel,
-                        panel.getBibDatabaseContext().getDatabaseFile()));
-            } catch (IOException ex) {
-                LOGGER.error("Problem registering file change notifications", ex);
-            }
-            frame.getFileHistory().newFile(panel.getBibDatabaseContext().getDatabaseFile().getPath());
+        File oldFile = panel.getBibDatabaseContext().getDatabaseFile();
```

Comment:
Oh, I prefer to not go into the Throwable-business... It seems to be easily replaceable, but there's a clear risk that eventually something will break...